ls1intum/Artemis
# `Communication`: Enable save button when multi-line content is copied into new posts

### Checklist
#### General



#### Client

### Motivation and Context
Previously while using the Discussion page, users were not able to save posts when they pasted the content part instead of typing, due to the 'Save' button not being enabled. Fixes #5244.

### Description
 After some debugging, we realized that the issue is present because the Regex pattern that verifies the validity of posting contents was not handling the edge case when the pasted text included the 'Carriage Return Character (\r)'. With this PR, we make sure that content with this escape character is also accepted by the posting content validator.

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Course with Postings enabled

1. Log in to Artemis
2. Navigate to Course Discussion page
3. Create or edit a Posting
4. Copy multiple lines 
5. Paste into the context of the Posting
6. Make sure that the "Save" button is enabled and postings can be saved.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
Not changed.

### Screenshots
https://user-images.githubusercontent.com/67344158/175520057-ab06f6be-699e-4da8-91dc-6b32e208534d.mp4
